src/formatDate.js

```diff
@@ -8,7 +8,48 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldYearIndex = getIndex(fromFormat);
+  const newYearIndex = getIndex(toFormat);
+  const oldYearLength = fromFormat[oldYearIndex].length;
+  const newYearLength = toFormat[newYearIndex].length;
+  const oldJoin = fromFormat[3];
```

Comment:
The variable names `oldJoin` and `newJoin` are not descriptive. It would be better to name them something like `oldSeparator` and `newSeparator` since they're specifically the separators in the date formats.